app/src/androidTest/java/com/eis/audioUtility/AudioUtilityManagerTest.java

```diff
@@ -0,0 +1,167 @@
+package com.eis.audioUtility;
+
+
+import android.content.Context;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author Francesco Bau', Alberto Ursino
+ */
+public class AudioUtilityManagerTest {
+
+    Context context;
+    private static final int INVALID_PERCENTAGE_LOW = -1;
+    private static final int INVALID_PERCENTAGE_HIGH = 101;
+
+    @Before
+    public void init() {
+        context = InstrumentationRegistry.getInstrumentation().getTargetContext();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void setAlarmVolume_percentage_isTooLow() {
+        AudioUtilityManager.setVolume(context, AudioUtilityManager.AUMStream.ALARM, INVALID_PERCENTAGE_LOW);
+        Assert.fail("Should have thrown IllegalArgumentException.");
```

Comment:
Little suggestion, since there's the `expected = IllegalArgumentException.class`, the `fail()` is redundant. This test will automatically fail if no exception is thrown and also if the wrong exception gets thrown

Reply:
You are totally right! I'm gonna fix it right away!